antarest/study/storage/rawstudy/model/filesystem/inode.py

```diff
@@ -124,6 +124,35 @@ def denormalize(self) -> None:
         """
         raise NotImplementedError()
 
+    def get_file_content(self) -> Tuple[bytes, str, str]:
+        """
+        Get file content
+
+        Returns:
+            file content, as bytes
+            file suffix
+            file name
+        """
+        suffix = self.config.path.suffix
+        filename = self.config.path.name
+        if self.config.archive_path:
+            return (
+                read_original_file_in_archive(self.config.archive_path, str(self.get_relative_path_inside_archive())),
+                suffix,
+                filename,
+            )
+        else:
+            return self.config.path.read_bytes(), suffix, filename
+
+    def get_relative_path_inside_archive(self) -> Path:
```

Comment:
I think you should give as an argument to this method the `archive_path`. This way you ensure we don't go inside the else and you can remove this Exception